### PR TITLE
qwen25_vl: fix Qwen2.5-VL-32B on-device decode gibberish on wh_llmbox_perf (#48037)

### PR DESCRIPTION
**Pull request by mtairum**
✅ **Fix validated on `wh_llmbox_perf` (T3000)** — run [28228897331](https://github.com/tenstorrent/tt-metal/actions/runs/28228897331): BERTScore **F1 = 0.790 > 0.75** and **`validate_perf_targets` PASS** (decode t/s/u within target). Full investigation log: #48037.

## Root cause
On-device greedy decode was running the **wrong sampling path**. On non-Galaxy hardware (incl. T3000) the sampling module is built with `default_sampling_force_argmax`, which sets `allow_force_argmax: False`. So greedy decode (temperature=0 → `format_sampling_params` normalizes to `k=1, p=0, temp=1`) fell through to the **heavy top-k/top-p multi-all-gather sampling pipeline** instead of the simple single-gather argmax path — and that heavy path corrupts at **batch-32** on T3000 (gibberish from ~token 2; correct at batch-1; host argmax of the same batch-32 run was correct, so the decode forward was always fine).

This was masked/compounded by two earlier regressions also surfaced here: #45522 (repeat-batch decode-input staleness) and the #45166 sampling-trace restructure.

## Fix (`models/demos/qwen25_vl/tt/model.py`, `models/tt_transformers/tt/generator.py`)
- **Enable `allow_force_argmax` for the qwen25_vl `Transformer`** (set in `SAMPLING_AG_CONFIG` before the sampling module is built) → greedy decode uses the single-gather argmax path (closest to the correct host argmax).
- `_tt_vllm_always_refresh_decode_trace_inputs = True` — re-stage decode trace inputs from host each step (perf-neutral; also fixes the #45522 repeat-batch staleness).
- `_tt_disable_sampling_trace = True` (+ `sample_decode_on_device` honoring it) — run the sampling op eagerly so the all-gather re-acquires a fresh `multi_device_global_semaphore` each step instead of reusing one frozen into a captured trace.
- Plus a `TT_QWEN_FORCE_HOST_SAMPLING` demo toggle (default off) kept as a known-good host-argmax reference for future A/B.

## Validation ledger (`wh_llmbox_perf`, target F1 > 0.75 **and** decode t/s/u ≥ target)
| run | commit | change | result |
|---|---|---|---|
| 28089820843 | `211176f` | gate #45522 page-table | F1 0.34 — disproven |
| 28116310815 | `69c6c73` | `Mode.PREFILL` reset | F1 0.338 — disproven |
| 28160314149 | `e74d655` | force host argmax | F1 0.791 ✅ / perf 18.4 ❌ → isolated to sampling |
| 28172123913 | `f9ad635` | always-refresh + on-device | perf 23.0 ✅ / gibberish ❌ → not input plumbing |
| 28194623853 | `6f629d15` | eager sampling | F1 0.362 ❌ → not the sampling-trace semaphore; batch-32-specific |
| 28228897331 | `a641d751` | **enable force-argmax (this fix)** | **F1 0.790 ✅ + perf PASS ✅ — GREEN** |

Note: the three flags are kept as the validated combination. A follow-up could test whether `allow_force_argmax` alone suffices (dropping the always-refresh / eager flags), but each check is a ~1.5h T3000 run; the combination is what's green.

Relates to #48037, #45522, #45166, #47218.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:mtairum/qwen25-vl-decode-pagetable-reset)

### CODE REVIEW

**Comment by mtairum** [comment]
Closing: the page-table-reset hypothesis is disproven (gating it made BERTScore F1 *worse*, 0.61→0.34 on hardware), and the bert-score cap this carried is now on `main` via #47802. The Qwen2.5-VL-32B gibberish is now tracked in **#48037** with the full bisect + ruled-out + low-level-suspect analysis. Branch left intact for reference.

File: models/demos/qwen25_vl/demo/demo.py [hunk 1]
```diff
@@ -510,7 +510,13 @@ def test_demo(
 
         # Start decoding
         iteration = 0
-        argmax_on_device = model._supports_on_device_sampling
+        # Diagnostic A/B toggle (#48037). The actual gibberish fix is the model-level
+        # `_tt_vllm_always_refresh_decode_trace_inputs = True` (Transformer in tt/model.py), which
+        # keeps on-device sampling. Setting TT_QWEN_FORCE_HOST_SAMPLING=1 forces host argmax
+        # instead: a known-good reference (correct output) but slower (per-step logits read-back),
+        # useful to compare against the on-device path locally.
```
[the remark this answers is not in the document]
Comment (mtairum) [reply models/demos/qwen25_vl/demo/demo.py]: Good catch — fixed in 96d612a. That comment predated the force-argmax fix and was stale. Updated it to correctly attribute the gibberish fix to enabling the on-device force-argmax path (`allow_force_argmax` via SAMPLING_AG_CONFIG, see tt/model.py); the ledger confirms always-refresh alone still produced gibberish (run 28172123913), and the green run 28228897331 is the force-argmax one.

File: models/tt_transformers/tt/generator.py [hunk 2]
```diff
@@ -1642,19 +1642,28 @@ def sample_decode_on_device(
...
                 else None
             )
             sampled_outputs.append(
                 sampling_module.sample(
                     logits=logits_i,
                     tt_out_tok=tt_out_tok,
-                    enable_trace=enable_trace,
+                    enable_trace=sampling_enable_trace,
                 )
             )
```
[the remark this answers is not in the document]
Comment (mtairum) [reply models/tt_transformers/tt/generator.py:1666]: Applied in 96d612a (collapsed to a single line, 116 chars).

**Review by yieldthought — APPROVED** [review]
